### DISCUSSION TITLE
Add example of converting Hasura graphql schema to Dart

### DISCUSSION THREAD

**Discussion by MohiuddinM** [hasura/graphql-engine#5956] [General] [answered]
I am using Hasura as a backend for a Flutter application. I have to write classes and queries manually, because existing generators don't work with Hasura's graphql schema. Can you please provide an example or documentation on how to generate classes and queries automatically for dart?

**Comment by Achudh** [comment]
> I am using Hasura as a backend for a Flutter application. I have to write classes and queries manually, because existing generators don't work with Hasura's graphql schema. Can you please provide an example or documentation on how to generate classes and queries automatically for dart?

2 years and still no answer. I was just running through this issue and came up with what can be done.
Possibly we could write a script which could read hasura metadata and tables in the db and create model for each table with the relations we get from hasura and make things nullable by default. The script could create a models folder under root or wherever and use those models when we parse data from graphql widgets. Just a thought

**Comment by meetzaveri** [comment, marked as answer]
Hello,

Apologies for the delay. We have submitted this request to our docs team. We will let you know.

In the meantime, we recommend checking [hasura_connnect](https://pub.dev/packages/hasura_connect) package from pub.dev which is made for facilitating Hasura integration with flutter. Also, there's an article on hasura blog on "How to get started with Hasura and flutter" - https://hasura.io/blog/getting-started-with-hasura-and-flutter/

**Reply by darkhorsejo** [reply]
The documentation for hasura_connect, in my view is poor and doesn't help new developer trying to understand the api to get anything done. Despite recommending hasura_connect plugin, however, your tutorial, from what I know, didn't cover how to use hasura_connect instead it focuses on using graphql_flutter and that tutorial is even out of date as it references an older version of graphql_flutter. If you are recommending hasura_connect could you please make a descent tutorial how to use the plugin?

**Reply by meetzaveri** [reply]
cc @catalinpit @praveenweb

**Comment by jtkeyva** [comment]
any updates? dart/flutter is skyrocketing and hasura + flutter is a great combo for 2023 tech

**Comment by wildsurfer** [comment]
Same problem in 2023

**Comment by oriollpz** [comment]
Still waiting

**Comment by darkhorsejo** [comment]
Done yet?
